cli/lsp/analysis.rs

```diff
@@ -135,9 +137,24 @@ fn as_lsp_range(range: &deno_lint::diagnostic::Range) -> Range {
 
 pub fn get_lint_references(
   parsed_source: &deno_ast::ParsedSource,
+  maybe_config_file: Option<&ConfigFile>,
 ) -> Result<Vec<Reference>, AnyError> {
   let syntax = deno_ast::get_syntax(parsed_source.media_type());
-  let lint_rules = rules::get_recommended_rules();
+  let maybe_lint_config = if let Some(config_file) = maybe_config_file {
```

Comment:
Hmmm, seeing this in action worries me a bit, because it means every time we `get_lint_references` we _re-parse_ the config file, and regenerate the rules, though we only need to do this once.

Maybe it would be better to:

- Add a `lint_rules` to the `Inner` of the language server.
- Calculate that once at the points like we do with with `update_tsconfig`, `update_import_map`, `update_cache` and `update_registries`.
- Include the `lint_rules` only in the snapshot.

I also am thinking this might be a better pattern, we don't want the interpretation of the config file dispersed through out the LSP.

Relocating this logic to only occur when a change occurs and in the language server would also make it easier to do a `self.client.show_message(MessageType::Warning, err).await;` when there is an error.  Just logging out an error to the debug output of the lsp won't really help people fix issues.

Reply:
`lint_rules` seems too specific to me. I think we should store parsed `FmtConfig` and `LintConfig` on the `Inner` and pass them down as needed.